modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java

```diff
@@ -262,6 +323,75 @@ private void _upgradeLinkToPageField(DDMFormField ddmFormField) {
 		ddmFormField.setType("link_to_layout");
 	}
 
+	private DDMForm _upgradeNestedFields(DDMForm ddmForm) {
+		if (!_hasNestedFields(ddmForm)) {
+			return ddmForm;
+		}
+
+		DDMForm newDDMForm = new DDMForm();
+
+		newDDMForm.setAvailableLocales(ddmForm.getAvailableLocales());
+		newDDMForm.setDefaultLocale(ddmForm.getDefaultLocale());
+		newDDMForm.setDefinitionSchemaVersion(
+			ddmForm.getDefinitionSchemaVersion());
+
+		for (DDMFormField ddmFormField : ddmForm.getDDMFormFields()) {
+			if (ListUtil.isEmpty(ddmFormField.getNestedDDMFormFields())) {
+				newDDMForm.addDDMFormField(ddmFormField);
+
+				continue;
+			}
+
+			DDMFormField fieldSetDDMFormField = _createFieldSetDDMFormField(
+				ddmForm.getDefaultLocale(), ddmFormField.getName() + "FieldSet",
+				ListUtil.fromArray(ddmFormField), ddmFormField.isRepeatable());
+
+			_upgradeNestedFields(
+				ddmFormField.getNestedDDMFormFields(),
+				ddmForm.getDefaultLocale(), newDDMForm, fieldSetDDMFormField);
+
+			fieldSetDDMFormField.setProperty(
+				"rows", _getDDMFormFieldsRows(fieldSetDDMFormField));
+
+			ddmFormField.setNestedDDMFormFields(Collections.emptyList());
+			ddmFormField.setRepeatable(false);
+
+			newDDMForm.addDDMFormField(fieldSetDDMFormField);
+		}
+
+		return newDDMForm;
+	}
+
+	private void _upgradeNestedFields(
+		List<DDMFormField> ddmFormFields, Locale defaultLocale,
+		DDMForm newDDMForm, DDMFormField parentFieldSetDDMFormField) {
```

Comment:
I believe that the newDDMForm parameter is not been used in this method, could you confirm, please?

Reply:
You are right it is no needed, I'm going to remove it